app/src/main/java/com/maq/xprize/onecourse/hindi/mainui/MainActivity.java

```diff
@@ -245,10 +237,16 @@ public void uncaughtException(Thread paramThread, Throwable paramThrowable) {
         }
     }
 
-    private boolean isExpansionExtractionRequired(int mainFileVersion, int patchFileVersion) {
+    private void startSplashScreenActivity() {
+        Intent intent = new Intent(MainActivity.this, SplashScreenActivity.class);
+        startActivity(intent);
+        finish();
+    }
+
+    private boolean isExpansionExtractionRequired(int storedMainFileVersion, int storedPatchFileVersion) {
         for (DownloadExpansionFile.XAPKFile xf : xAPKS) {
             // If main or patch file is updated set isExtractionRequired to true
-            if (xf.mIsMain && xf.mFileVersion != mainFileVersion || !xf.mIsMain && xf.mFileVersion != patchFileVersion) {
+            if (xf.mIsMain && xf.mFileVersion != storedMainFileVersion || !xf.mIsMain && xf.mFileVersion != storedPatchFileVersion) {
```

Comment:
Shouldn't the patch version be checked separately? I mean the patch version can be different than the main version.